lib/rspec/support/object_inspector.rb

```diff
@@ -13,6 +13,8 @@ def self.inspect(object)
           "#{object.to_s 'F'} (#{object.inspect})"
         elsif RSpec::Support.is_a_matcher?(object) && object.respond_to?(:description)
           object.description
+        elsif object.class == RSpec::Mocks::Double
+          "Double (#{object.instance_variable_get(:@name)})"
```

Comment:
Hmm I feel uncomfortable prying into the object like this, but exposing name directly would be worse...